### ISSUE TITLE
Allow a "createClient" method to be passed in

### ISSUE THREAD

**Issue by richthegeek** [OptimalBits/bull#124]
This is mostly in relation to #106 but is slightly more general as there have been a few cases whereby I've not easily had the full Redis options available in the section of the codebase where Bull is being initialised.

So my proposal would be that a "createClient" parameter can be passed into the Queue as an option, such that you can call "sentinel.createClient()" or similar as required.

So the change to the Queue constructor is adding a "createClient" param at the end (and as part of the object-method of initialising), and then having something like:

```
  createClient = createClient || function (port, host, options) {
    return redis.createClient(port, host, options);
  }

  this.client = createClient(redisPort, redisHost, redisOptions);
```

We are actually running this in dev at the moment, so I can push a PR if you think this is a good idea?

**Comment by richthegeek** [comment]
https://github.com/richthegeek/bull/commit/7d00dec189a05dfcbdf97a219d9abb71e6719c91

That's a working version of the above idea. Note that it needs to be promisified if someone is otherwise passing in a native node-redis object.

**Comment by richthegeek** [comment]
Oh wow this is so similar to #121 haha...

**Comment by manast** [comment]
fixed in master: 20d48bdd83dbcd2741e25d63ad3b72bc225e458f
